extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudStreamExtractor.java

```diff
@@ -185,43 +187,102 @@ public String getHlsUrl() {
         try {
             final JsonArray transcodings = track.getObject("media").getArray("transcodings");
 
-            // get information about what stream formats are available
-            for (Object transcoding : transcodings) {
+            // Iterate a first time to see if there is a progressive MP3 stream available.
+            // If yes, the MP3 HLS stream will be not added to audioStreams.
 
+            boolean mp3ProgressiveStreamInTranscodings = false;
+
+            for (final Object transcoding : transcodings) {
+                final JsonObject t = (JsonObject) transcoding;
+                if (t.getString("preset").contains("mp3") &&
+                    t.getObject("format").getString("protocol").equals("progressive")) {
+                    mp3ProgressiveStreamInTranscodings = true;
+                }
+            }
+
+            // Get information about what stream formats are available
+            for (final Object transcoding : transcodings) {
                 final JsonObject t = (JsonObject) transcoding;
                 String url = t.getString("url");
+                final String mediaUrl;
+                final MediaFormat mediaFormat;
+                final int bitrate;
 
                 if (!isNullOrEmpty(url)) {
+                    if (t.getString("preset").contains("mp3")) {
+                        // Don't add the MP3 HLS stream if there is a progressive stream present
+                        // because the two have the same bitrate
+                        if (t.getObject("format").getString("protocol").equals("hls") &&
+                            mp3ProgressiveStreamInTranscodings) {
```

Comment:
Reversing the order of the two boolean should give a very very minor performance boost, since `mp3ProgressiveStreamInTranscodings` is already calculated